packages/odyssey-react-mui/src/MenuItem.tsx

```diff
@@ -10,40 +10,63 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { MenuItem as MuiMenuItem } from "@mui/material";
+import {
+  MenuItem as MuiMenuItem,
+  MenuItemProps as MuiMenuItemProps,
+} from "@mui/material";
 import { menuItemClasses } from "@mui/material/MenuItem";
-import type { MenuItemProps as MuiMenuItemProps } from "@mui/material";
-import { memo, useContext } from "react";
+import { ReactNode, memo, useCallback, useContext } from "react";
 
 import { MenuContext } from "./MenuContext";
 
-export interface MenuItemProps
-  extends Omit<
-    MuiMenuItemProps,
-    | "component"
-    | "dense"
-    | "disableGutters"
-    | "divider"
-    | "focusVisibleClassName"
-  > {
-  /**
-   * Toggles whether or not the MenuItem represents a destructive action.
-   */
+export type MenuItemProps = {
+  children: ReactNode;
+  content?: string;
+  hasInitialFocus?: boolean;
+  isSelected?: boolean;
   isDestructive?: boolean;
-}
+  onClick?: MuiMenuItemProps["onClick"];
+  rel?: string;
+  rev?: string;
+  value?: string;
+};
```

Comment:
Can we add docs for these props? I specifically wanna know the `rel` and `rev` ones as they're not clear.

`isDestructive` is also unclear. Does that just turn it red? it doesn't change any functionality? Because at that point I'd consider it a `"variant"` of the component rather than a state.

Reply:
`isDestructive` is also unclear. Does that just turn it red? it doesn't change any functionality? Because at that point I'd consider it a `"variant"` of the component rather than a state.

Reply:
What's the difference between `children` and `content`? I know it's a MUI thing, but I can't tell from looking at the props.

Reply:
- I removed `rel` because it only applies if it's a link with an `href`, which `MenuItem` should never be
- I removed `rev` because it hasn't been an acceptable attribute since HTML 4 (yikes)
- I removed `content` because I have no idea what it does, can't find documentation in MUI, and can't find any legitimate use cases
- I changed `isDestructive` to be a `destructive` variant